### PR TITLE
Making addIndexedMappings more stable

### LINKED ISSUE #4: Crash: napi_create_reference
https://github.com/parcel-bundler/parcel/pull/4313/checks?check_run_id=543074718#step:3:366

```
FATAL ERROR: Error::Error napi_create_reference
 1: 0x9bcb80 node::Abort() [/usr/local/bin/node]
 2: 0x9bdd16 node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
 3: 0x9bde09  [/usr/local/bin/node]
 4: 0x99600b napi_fatal_error [/usr/local/bin/node]
 5: 0x7f4f742c75b6  [/usr/src/app/.tmp/parcel-v2/node_modules/@parcel/source-map/build/Release/sourcemap.node]
 6: 0x7f4f742c780f Napi::Error::New(napi_env__*) [/usr/src/app/.tmp/parcel-v2/node_modules/@parcel/source-map/build/Release/sourcemap.node]
 7: 0x7f4f742c8627 Napi::Object::Get(char const*) const [/usr/src/app/.tmp/parcel-v2/node_modules/@parcel/source-map/build/Release/sourcemap.node]
 8: 0x7f4f742bee02 SourceMapBinding::addIndexedMappings(Napi::CallbackInfo const&) [/usr/src/app/.tmp/parcel-v2/node_modules/@parcel/source-map/build/Release/sourcemap.node]
 9: 0x7f4f742c8d98 Napi::ObjectWrap<SourceMapBinding>::InstanceVoidMethodCallbackWrapper(napi_env__*, napi_callback_info__*) [/usr/src/app/.tmp/parcel-v2/node_modules/@parcel/source-map/build/Release/sourcemap.node]
10: 0x9770e5  [/usr/local/bin/node]
11: 0xb8466c  [/usr/local/bin/node]
12: 0xb86477 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/usr/local/bin/node]
13: 0x134e979  [/usr/local/bin/node]
Aborted (core dumped)
```

It didn't happen locally for me so far.

Maybe this happens because the `Get()` calls here
https://github.com/parcel-bundler/source-map/blob/44b5a71540a0fca47420952fee2faf734671d41b/src/SourceMap.cpp#L409-L424
throw when the specified property doesn't exist.

### PR DESCRIPTION
**Pull request by DeMoorJasper**
This fixes potential problems with `addIndexedMappings` which cannot really handle invalid mappings at this point.

This PR adds some extra tests and checks to ensure it does not crash.

Closes #4

### CODE REVIEW

**Comment by mischnic** [comment]
So should we throw an error on invalid mappings rather than skipping them?

**Comment by DeMoorJasper** [comment]
@mischnic not sure, we just filtered them out before as we didn't wanna fail builds for tools that don't know how to make sourcemaps

**Comment by DeMoorJasper** [comment]
@mischnic I've added a benchmark and got very surprising results, it's quicker to add rawmappings to mozilla's source-map, stringify it and import those vlq mappings than it is to just add indexedmappings straight into `@parcel/source-map`.

I'll experiment a bit further in here with maybe an `addIndexedMapping`

File: src/MappingContainer.cpp [hunk 1]
```diff
@@ -437,3 +437,22 @@ void MappingContainer::addBufferMappings(const void *buf, int lineOffset, int co
...
 }
+
+void MappingContainer::addIndexedMapping(int generatedLine, int generatedColumn, int originalLine, int originalColumn,
+                                         std::string source, std::string name) {
+    Position generatedPosition = Position{generatedLine, generatedColumn};
+    Position originalPosition = Position{originalLine, originalColumn};
+    int sourceIndex = -1;
+    int nameIndex = -1;
+    if (originalPosition.line > -1) {
+        if (source.size() > 0) {
```
Comment (mischnic) [inline src/MappingContainer.cpp:448]:
```suggestion
        if (!source.empty()) {
```

File: src/MappingContainer.cpp [hunk 2]
```diff
@@ -437,3 +437,22 @@ void MappingContainer::addBufferMappings(const void *buf, int lineOffset, int co
...
+            sourceIndex = addSource(source);
+        }
+
+        if (name.size() > 0) {
```
Comment (mischnic) [inline src/MappingContainer.cpp:452]:
```suggestion
        if (!name.empty()) {
```

**Pull request merged**